tests/gauge/test_boost_delegation.py

```diff
@@ -0,0 +1,24 @@
+from brownie import ETH_ADDRESS
+
+
+def test_boost_delegation_not_set_in_proxy(alice, gauge, voting_escrow):
+    tx = gauge.user_checkpoint(alice, {"from": alice})
+    subcall = next(
+        (tx for tx in tx.subcalls if tx["function"] == "adjusted_balance_of(address)"), {}
+    )
+    next_subcall = tx.subcalls[tx.subcalls.index(subcall) + 1]
+
+    expected = {"to": voting_escrow.address, "op": "STATICCALL", "function": "balanceOf(address)"}
+    for k in expected.keys():
+        assert next_subcall[k] == expected[k]
+
+
+def test_boost_delegation_set_in_addr_provider(alice, gauge, mock_veboost_proxy):
+    mock_veboost_proxy.set_delegation(ETH_ADDRESS, {"from": alice})
+
+    tx = gauge.user_checkpoint(alice, {"from": alice})
+    subcall = next(
+        (tx for tx in tx.subcalls if tx["function"] == "adjusted_balance_of(address)"), {}
+    )
+
+    assert subcall["return_value"][0] == 69
```

Comment:
here we aren't testing anything other than the fact that the proxy is called. And for now we know that two behaviors exist, the first is if the proxy points to the veboost contract we will get the adjusted veboost, the second being if the delegation is not set, we should get the plain voting escrow balance of a user.

Those however are implementation details of the voting escrow boost contract itself.